powerauth-restful-security-javaee/src/main/java/io/getlime/security/powerauth/rest/api/jaxrs/controller/v3/UpgradeController.java

```diff
@@ -126,12 +127,13 @@ public Response upgradeCommit(@HeaderParam(PowerAuthSignatureHttpHeader.HEADER_N
         try {
             PowerAuthSignatureHttpHeaderValidator.validate(header);
         } catch (InvalidPowerAuthHttpHeaderException ex) {
-            throw new PowerAuthUpgradeException(ex.getMessage());
+            logger.warn("Signature validation failed, error: {}", ex.getMessage());
+            throw new PowerAuthUpgradeException();
```

Comment:
Let's pass `ex` as a parameter to the new exception to keep the cause somewhere.